### PR TITLE
fix(v3): Add optimizations to system table queries #5576

### LINKED ISSUE #5576: Add optimizations to system table queries
Related to https://github.com/influxdata/influxdb_iox/issues/11974

Add information related to optimizing to system table queries to prevent these queries from affecting the overall performance of the cluster. The examples that do not include `WHERE` clauses should probably be removed.

##### Relevant URLs
- https://docs.influxdata.com/influxdb/cloud-dedicated/admin/query-system-data/

**Note**: Also need to port this doc to the InfluxDB Clustered documentation.

### PR DESCRIPTION
**Pull request by jstirnaman**
- Closes #5576
- Integrates guidance from https://github.com/influxdata/influxdb_iox/issues/11974 for Cloud Dedicated and Clustered.
- Updates examples to include filter conditions.

### CODE REVIEW

File: content/influxdb/cloud-dedicated/admin/query-system-data.md [hunk 1]
```diff
@@ -119,6 +128,112 @@ Replace the following:
...
+Use the following filters to optimize your system table queries and reduce the impact on your
+cluster's performance.
+
+##### Filter by table name
+
+Filter by `table_name` when querying `system.tables`, `system.partitions`, or `system.compactor`. 
+
+```sql
+WHERE table_name = 'TABLE_NAME' (string)
+```
```
Comment (sanderson) [inline content/influxdb/cloud-dedicated/admin/query-system-data.md:149]:
~~~suggestion
When querying the `system.tables`, `system.partitions`, or `system.compactor` tables, use the
`WHERE` clause to filter by `table_name` . 

```sql
SELECT * FROM system.partitions WHERE table_name = 'TABLE_NAME'
```
~~~

File: content/influxdb/cloud-dedicated/admin/query-system-data.md [hunk 2]
```diff
@@ -119,6 +128,112 @@ Replace the following:
...
+
+##### Filter by partition key 
+
+Filter by `partition_key` when querying `system.partitions` or `system.compactor`.
+
+```sql
+WHERE partition_key = 'PARTITION_KEY (string)
+```
```
Comment (sanderson) [inline content/influxdb/cloud-dedicated/admin/query-system-data.md:157]:
~~~suggestion
When querying the `system.partitions` or `system.compactor` tables, use the `WHERE` clause to
filter by `partition_key` .

```sql
SELECT * FROM system.partitions WHERE partition_key = 'PARTITION_KEY'
```
~~~

File: content/influxdb/cloud-dedicated/admin/query-system-data.md [hunk 3]
```diff
@@ -119,6 +128,112 @@ Replace the following:
...
+
+To further improve performance, use `AND` to pair `partition_key` with `table_name`--for example: 
+   
+```sql
+SELECT * 
+FROM system.partitions 
+WHERE table_name = 'TABLE_NAME' 
+ AND partition_key = 'PARTITION_KEY';
+```
```
Comment (sanderson) [inline content/influxdb/cloud-dedicated/admin/query-system-data.md:166]:
~~~suggestion
```sql
SELECT * 
FROM system.partitions 
WHERE
  table_name = 'TABLE_NAME' 
  AND partition_key = 'PARTITION_KEY';
```
~~~

File: content/influxdb/cloud-dedicated/admin/query-system-data.md [hunk 4]
```diff
@@ -119,6 +128,112 @@ Replace the following:
...
+
+- `PARTITION_KEY`: a [partition key]((/influxdb/cloud-dedicated/admin/custom-partitions/#partition-keys))
+   derived from the table's partition template.
```
Comment (sanderson) [inline content/influxdb/cloud-dedicated/admin/query-system-data.md:169]: Are these meant to be code-placeholders?

File: content/influxdb/cloud-dedicated/admin/query-system-data.md [hunk 5]
```diff
@@ -119,6 +128,112 @@ Replace the following:
...
+The default format for `partition_key` is `%Y-%m-%d` (for example, `2024-01-01`).
+If the table uses a custom partition template, the `partition_key` format will differ.
+
+##### Filter by partition ID 
+
+Filter by `partition_id` when querying `system.partitions` or `system.compactor`.
+
+```sql
+WHERE partition_id = PARTITION_ID (int64)
+```
```
Comment (sanderson) [inline content/influxdb/cloud-dedicated/admin/query-system-data.md:182]:
~~~suggestion
When querying the `system.partitions` or `system.compactor` table, use the `WHERE` clause to
filter by `partition_id` .

```sql
SELECT * FROM system.partitions WHERE partition_id = PARTITION_ID
```
~~~

File: content/influxdb/cloud-dedicated/admin/query-system-data.md [hunk 6]
```diff
@@ -119,6 +128,112 @@ Replace the following:
...
+
+For the most optimized approach, use `AND` to pair `partition_id` with `table_name`--for example:
+
+```sql
+SELECT * 
+FROM system.partitions 
+WHERE table_name = 'TABLE_NAME' 
+ AND partition_id = PARTITION_ID;
+```
```
Comment (sanderson) [inline content/influxdb/cloud-dedicated/admin/query-system-data.md:191]:
~~~suggestion
```sql
SELECT * 
FROM system.partitions 
WHERE
  table_name = 'TABLE_NAME' 
  AND partition_id = PARTITION_ID;
```
~~~

File: content/influxdb/cloud-dedicated/admin/query-system-data.md [hunk 7]
```diff
@@ -119,6 +128,112 @@ Replace the following:
...
+
+- `PARTITION_ID`: a [partition ID](#retrieve-a-partition-id)
```
Comment (sanderson) [inline content/influxdb/cloud-dedicated/admin/query-system-data.md:193]:
```suggestion
- `PARTITION_ID`: a [partition ID](#retrieve-a-partition-id) (int64)
```

File: content/influxdb/cloud-dedicated/admin/query-system-data.md [hunk 8]
```diff
@@ -119,6 +128,112 @@ Replace the following:
...
+###### Retrieve a partition ID
+
+To retrieve a partition ID, query `system.partitions` for a `table_name` and `partition_key` pair--for example:
+
+```sql
+SELECT * 
+FROM system.partitions
+WHERE table_name = 'TABLE_NAME'
+  AND partition_key = 'PARTITION_KEY';
+```
```
Comment (sanderson) [inline content/influxdb/cloud-dedicated/admin/query-system-data.md:207]:
~~~suggestion
```sql
SELECT
  table_name,
  partition_key,
  partition_id 
FROM system.partitions
WHERE
  table_name = 'TABLE_NAME'
  AND partition_key = 'PARTITION_KEY';
```
~~~

File: content/influxdb/cloud-dedicated/admin/query-system-data.md [hunk 9]
```diff
@@ -119,6 +128,112 @@ Replace the following:
...
+
+The result contains the `partition_id`:
+
+| partition_id | table_name | partition_key     | last_new_file_created_at | num_files | total_size_mb |
+| -----------: | :--------- | :---------------- | -----------------------: | --------: | ------------: |
+|         1362 | weather    | 43 \| 2020-05-27  |      1683747418763813713 |         1 |             0 |
```
Comment (sanderson) [inline content/influxdb/cloud-dedicated/admin/query-system-data.md:213]:
```suggestion
The result contains the `partition_id`:

| table_name | partition_key     | partition_id |
| :--------- | :---------------- | -----------: |
| weather    | 43 \| 2020-05-27  |         1362 |
```

**Review by sanderson — CHANGES_REQUESTED** [review]
Suggestions to the Dedicated version. Once applied, just port them to the Clustered version. (I didn't realize I hadn't ported this doc over to Clustered, thanks for doing that).

**Review by sanderson — APPROVED** [review]
Looks great! Thanks @Jason. One minor suggestion, but I'm not going to hold the PR up for it.

File: content/influxdb/cloud-dedicated/admin/query-system-data.md [hunk 10]
```diff
@@ -119,6 +128,120 @@ Replace the following:
...
+WHERE
+  table_name = 'TABLE_NAME'
+  AND partition_key = 'PARTITION_KEY';
+```
+
+The result contains the `partition_id`:
+
+| table_name | partition_key     | partition_id |
+| :--------- | :---------------- | -----------: |
+| weather    | 43 \| 2020-05-27  |         1362 |
```
Comment (appletreeisyellow) [inline content/influxdb/cloud-dedicated/admin/query-system-data.md]: I like this example -- just select three columns to show only the necessary information. Neat! 🧹

File: content/influxdb/cloud-dedicated/admin/query-system-data.md [hunk 11]
```diff
@@ -359,9 +502,11 @@ SELECT
   SUM(total_l0_bytes) + SUM(total_l1_bytes) + SUM(total_l2_bytes) AS total_bytes
 FROM
   system.compactor
+WHERE
+  table_name = 'TABLE_NAME'
```
Comment (appletreeisyellow) [inline content/influxdb/cloud-dedicated/admin/query-system-data.md:517]: Maybe also add something like `for a specific table` at the end of the example title -- i.e. `View the total size of all compacted partitions in bytes for a specific table`

File: content/influxdb/cloud-dedicated/admin/custom-partitions/_index.md [hunk 12]
```diff
@@ -74,6 +74,7 @@ Custom partitioning has the following limitations:
 - Database and table partitions can only be defined on create.
   You cannot update the partition strategy of a database or table after it has
   been created.
+- A partition template must include a time interval.
```
Comment (appletreeisyellow) [inline content/influxdb/cloud-dedicated/admin/custom-partitions/_index.md:77]:
```suggestion
- A partition template must include a time part.
```

**Review by appletreeisyellow — APPROVED** [review]
Thank you @jstirnaman for putting them together! Very nice work 💯

I left some non-blocking suggestions

**Pull request merged**